src/signature_verifier.rs

```diff
@@ -1,34 +1,35 @@
 use crate::models::SlackSigningSecret;
-use ring::hmac;
+use hmac::{Hmac, Mac};
 use rsb_derive::Builder;
 use rvstruct::*;
+use sha2::Sha256;
 use std::error::Error;
 use std::fmt::{Display, Formatter};
+use subtle::ConstantTimeEq;
 
 #[derive(Clone)]
 pub struct SlackEventSignatureVerifier {
-    secret_len: usize,
-    key: hmac::Key,
+    secret_bytes: Vec<u8>,
 }
 
 impl SlackEventSignatureVerifier {
     pub const SLACK_SIGNED_HASH_HEADER: &'static str = "x-slack-signature";
     pub const SLACK_SIGNED_TIMESTAMP: &'static str = "x-slack-request-timestamp";
 
     pub fn new(secret: &SlackSigningSecret) -> Self {
-        let secret_bytes = secret.value().as_bytes();
-        SlackEventSignatureVerifier {
-            secret_len: secret_bytes.len(),
-            key: hmac::Key::new(hmac::HMAC_SHA256, secret_bytes),
-        }
+        let secret_bytes = secret.value().as_bytes().to_vec();
+        SlackEventSignatureVerifier { secret_bytes }
     }
 
     fn sign<'a, 'b>(&'a self, body: &'b str, ts: &'b str) -> String {
-        let data_to_sign = format!("v0:{ts}:{body}");
-        format!(
-            "v0={}",
-            hex::encode(hmac::sign(&self.key, data_to_sign.as_bytes()))
-        )
+        let mut mac = Hmac::<Sha256>::new_from_slice(&self.secret_bytes)
+            .expect("HMAC can take key of any size");
```

Comment:
Nit: I'd prefer to avoid panicking here. we can just reuse existing error, I think we have something for system errors.